### PR TITLE
[MOS-425] Crash fix on unchecked nullptr

### PR DESCRIPTION
**Pull request by pholat**
Crash fix 

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

**Review by Maciej-Mudita — APPROVED** [review]

**Review by SP2FET — APPROVED** [review]

File: module-services/service-db/ServiceDBCommon.cpp [hunk 1]
```diff
@@ -35,10 +35,18 @@ sys::MessagePointer ServiceDBCommon::DataReceivedHandler(sys::DataMessage *msgl,
...
-        auto recordID  = result->getRecordID().value_or(0);
-        responseMsg    = std::make_shared<db::QueryResponse>(std::move(result));
+        std::optional<uint32_t> val;
+        if (result != nullptr) {
+            val = result->getRecordID();
+        }
+        else {
+            LOG_WARN("There is no response associated with query: %s!", query ? query->debugInfo().c_str() : "");
+        }
+        responseMsg = std::make_shared<db::QueryResponse>(std::move(result));
```
Comment (swichu91) [inline module-services/service-db/ServiceDBCommon.cpp:45]: Do we want to pass `nullptr` to `db::QueryResponse`?

File: module-services/service-db/ServiceDBCommon.cpp [hunk 2]
```diff
@@ -35,10 +35,18 @@ sys::MessagePointer ServiceDBCommon::DataReceivedHandler(sys::DataMessage *msgl,
...
 
-        sendUpdateNotification(msg->getInterface(), queryType, recordID);
+        if (val) {
+            sendUpdateNotification(msg->getInterface(), queryType, val.value());
```
Comment (swichu91) [inline module-services/service-db/ServiceDBCommon.cpp:48]: Calling `val.value()` looks a bit strange to me. How about replacing it with `*val` or changing the name of the `val` variable to something more adequate?

File: module-services/service-db/ServiceDBCommon.cpp (hunk 1 above)
Comment (pholat) [reply module-services/service-db/ServiceDBCommon.cpp:45]: Yes, I didn't want to change logic here while introducing the fix..

File: module-services/service-db/ServiceDBCommon.cpp (hunk 2 above)
Comment (pholat) [reply module-services/service-db/ServiceDBCommon.cpp:48]: That's true. I'll rename it?

**Review by swichu91 — APPROVED** [review]

**Pull request merged**